Panic observed when removing multiple stakes during sortition

When removing multiple stakes (or any stake besides the very last one) from one provisioner, the current `range` loop will go out of bounds:

```
panic: runtime error: index out of range [2] with length 2

goroutine 54 [running]:
github.com/dusk-network/dusk-blockchain/pkg/core/consensus/user.(*Member).RemoveStake(...)
        /home/jules/go/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/user/provisioners.go:38
github.com/dusk-network/dusk-blockchain/pkg/core/consensus/user.Provisioners.CreateVotingCommittee(0xc0012ac400, 0x30, 0x40, 0xc0033e5320, 0x3d035, 0x2, 0x30, 0x5d46e2, 0xc0021d9c88, 0xc002dd2f30, ...)
        /home/jules/go/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/user/sortition.go:82 +0x4d5
github.com/dusk-network/dusk-blockchain/pkg/core/verifiers.checkBlockCertificateForStep(0xc0021d9c78, 0xedf6d7f, 0x3d035, 0xc0021d9c02, 0xc0012ac400, 0x30, 0x40, 0xc000169500, 0xc00269c6e0, 0x20, ...)
        /home/jules/go/src/github.com/dusk-network/dusk-blockchain/pkg/core/verifiers/block.go:84 +0xce
github.com/dusk-network/dusk-blockchain/pkg/core/verifiers.CheckBlockCertificate(0xc0012ac400, 0x30, 0x40, 0xc000169500, 0xc0000f3980, 0xc00094d600, 0x20, 0x20, 0x20, 0x20)
        /home/jules/go/src/github.com/dusk-network/dusk-blockchain/pkg/core/verifiers/block.go:75 +0x158
github.com/dusk-network/dusk-blockchain/pkg/core/chain.(*Chain).AcceptBlock(0xc00015a300, 0xc0000f3980, 0xc00094d600, 0x20, 0x20, 0x0, 0x0)
        /home/jules/go/src/github.com/dusk-network/dusk-blockchain/pkg/core/chain/chain.go:269 +0x52f
github.com/dusk-network/dusk-blockchain/pkg/core/chain.(*Chain).finalizeIntermediateBlock(...)
        /home/jules/go/src/github.com/dusk-network/dusk-blockchain/pkg/core/chain/chain.go:628
github.com/dusk-network/dusk-blockchain/pkg/core/chain.(*Chain).handleCertificateMessage(0xc00015a300, 0xc00131ad60, 0x20, 0x20, 0xc0001b9590)
        /home/jules/go/src/github.com/dusk-network/dusk-blockchain/pkg/core/chain/chain.go:607 +0x61a
github.com/dusk-network/dusk-blockchain/pkg/core/chain.(*Chain).Listen(0xc00015a300)
        /home/jules/go/src/github.com/dusk-network/dusk-blockchain/pkg/core/chain/chain.go:153 +0x50e
created by main.Setup
        /home/jules/go/src/github.com/dusk-network/dusk-blockchain/launch/testnet/server.go:58 +0x2d9
```

I've already been able to reproduce it in a test. A very simple fix is to adjust the loop so that the index only increments if no stake is removed. A fix will be pushed shortly.

--- a/pkg/core/consensus/user/sortition.go
+++ b/pkg/core/consensus/user/sortition.go
@@ -77,11 +77,19 @@ func (p Provisioners) CreateVotingCommittee(round uint64, step uint8, size int) 
 
 	// Remove stakes which have not yet become active, or have expired
 	for _, m := range p.Members {
-		for i, stake := range m.Stakes {
-			if stake.StartHeight > round || stake.EndHeight < round {
-				subtractFromTotalWeight(W, stake.Amount)
-				m.RemoveStake(i)
+		i := 0
+		for {
+			if i == len(m.Stakes) {
+				break
 			}
+
+			if m.Stakes[i].StartHeight > round || m.Stakes[i].EndHeight < round {
+				subtractFromTotalWeight(W, m.Stakes[i].Amount)
+				m.RemoveStake(i)
+				continue
+			}
+
+			i++
 		}
 	}
 

--- a/pkg/core/consensus/user/sortition_test.go
+++ b/pkg/core/consensus/user/sortition_test.go
@@ -73,3 +73,15 @@ func TestSubtractStake(t *testing.T) {
 		assert.Equal(t, uint64(500), stake)
 	}
 }
+
+// Test that removing a stake from the array can not cause a panic
+func TestRemoveStake(t *testing.T) {
+	p, _ := consensus.MockProvisioners(10)
+
+	// Let's add two stakes which are not yet active to one of the provisioners.
+	p.MemberAt(2).AddStake(user.Stake{500, 1000, 10000})
+	p.MemberAt(2).AddStake(user.Stake{500, 1000, 10000})
+
+	// Now, extract a committee for round 1 step 1
+	assert.NotPanics(t, func() { p.CreateVotingCommittee(1, 1, 10) })
+}

